### ISSUE TITLE
Thin Logical Volume size - use entire VG

### ISSUE THREAD

**Issue by holmesb** [gluster/gluster-ansible-infra#74]
I'm trying to create a brick that uses the entire physical disk via a thin Logical Volume, without specifying an absolute size (in GB\TB, etc).

The thin pool is fine since can use:
`thinpoolsize: "99%VG"`, which leaves one percent for the pool metadata (16GB).  

But this approach fails when creating the thin LV using variable: `gluster_infra_lv_logicalvols` that has: `lvsize:  "99%VG"`.   Error occurs: "Thin volume sizing with percentage not supported":
```
ASK [gluster.infra/roles/backend_setup : Create thin logical volume] *************************************************************************************************************************************************************************

Tuesday 29 October 2019  15:38:19 +0000 (0:00:00.103)       0:00:29.366 *******
failed: [storage-server] (item={'vgname': 'vg_sda', 'thinpool': 'vg_sda_thinpool', 'lvname': 'vg_sda_thinlv', 'lvsize': '99%VG'}) => {"changed": false, "index": 0, "item": {"lvname": "vg_sda_thinlv", "lvsize": "99%VG", "thinpool": "vg_sda_thinpool", "vgname": "vg_sda"}, "msg": "Thin volume sizing with percentage not supported."}
```
How can I avoid specifying the absolute size of each disk in the `lvsize` value?

**Comment by holmesb** [comment]
"by pre-allocating the entire disk's space in a thin Logical Volume" - why not use a thick volume instead?  We need to use snapshots, for which thin volumes are a pre-requisite.

**Comment by sac** [comment]
Under thinpool and thinvolume create if you don't give size 100% will be assumed. 

roles/backend_setup/tasks/thin_pool_create.yml:     size: "{{ item.thinpoolsize | default('100%FREE') }}"

Leave out the thinpoolsize parameter. It will use the entire disk.
Refer: https://github.com/gluster/gluster-ansible/blob/master/playbooks/hc-ansible-deployment/gluster_inventory.yml for an example.

**Comment by holmesb** [comment]
I take your point for `gluster_infra_thinpools.thinpoolsize`, but sizing for `gluster_infra_thinpools.poolmetadatasize` isn't consistent.  [`poolmetadatasize` has a default in the "Create a LV thinpool-meta"](https://github.com/gluster/gluster-ansible-infra/blob/e1f54b4d7583878f2b9823c6d8b39b6106e553e2/roles/backend_setup/tasks/thin_pool_create.yml#L157), but in the same task: ["Create a thin pool" will error ](https://github.com/gluster/gluster-ansible-infra/blob/e1f54b4d7583878f2b9823c6d8b39b6106e553e2/roles/backend_setup/tasks/thin_pool_create.yml#L227)if a value isn't specified.  

I think the original issue I raised here is with the [lvol module ](https://github.com/ganto/ansible/blob/7aabdca2d6de72041b569dbf29fb19f44eb6f3d3/lib/ansible/modules/system/lvol.py#L436)rather than this role.  It doesn't support [percent values for size](https://github.com/gluster/gluster-ansible-infra/blob/7c3b8a31d6924b3ae96c2f8ae444cbc0f90dfcac/roles/backend_setup/tasks/thin_volume_create.yml#L138) when creating thin volumes.

**Comment by sac** [comment]
You are right regarding the consistency. But poolmetadata size we cannot leave it to default to 100%. Since it is just the metadata. And we have to default to a saner value if it is not given than erroring out. Because we know what would be the ideal value for metadata.

Regarding lvol module, you are right. It is not quite flexible.
